evennia/objects/objects.py

```diff
@@ -959,7 +959,22 @@ def find_clone_key():
                       if obj.key.startswith(key) and obj.key.lstrip(key).isdigit())
             return "%s%03i" % (key, num)
         new_key = new_key or find_clone_key()
-        return ObjectDB.objects.copy_object(self, new_key=new_key)
+        new_obj = ObjectDB.objects.copy_object(self, new_key=new_key)
+        new_obj.at_object_creation_copy(self)
+        return new_obj
+
+    def at_object_creation_copy(self, source_obj):
```

Comment:
- I think the name of this hook should be `at_post_copy` since it's called by the `copy` hook, after the copy has been performed. 
- Since this hook is called from within the `source_object.copy` method, it makes more sense to call it on the `source_object` instead and to pass the `copy` in as an argument. You have full flexibility to customize things either way.